extensions/azurecore/src/utils/Logger.ts

```diff
@@ -30,3 +48,52 @@ export class Logger {
 		return this._piiLogging;
 	}
 }
+
+/**
+ * Sanitizes a given object for logging to the output window, removing/shortening any PII or unneeded values
+ * @param objOrArray The object to sanitize for output logging
+ * @returns The stringified version of the sanitized object
+ */
+function sanitize(objOrArray: any): string {
+	if (Array.isArray(objOrArray)) {
```

Comment:
A further improvement could be to have a deepSanitize that walks the object and sanitizes every object it comes across. Something to keep in mind if we find the current implementation inadequate for all scenarios.